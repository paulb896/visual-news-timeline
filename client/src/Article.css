.article {
    outline: solid rgba(50, 100, 100, 0.2) 1px;
    display: inline-grid;
    padding: 20px;
    margin: 8px;
    text-align: justify;
    width: 550px;
    justify-content: center;
}