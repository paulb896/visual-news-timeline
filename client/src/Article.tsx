import * as React from 'react';

import './Article.css';

class Article extends React.Component<any, any> {

    constructor(props: any) {
        super(props);
        const { articleBody, headline, id, publishedAt, url } = props;

        this.state = {
            articleBody,
            headline,
            id,
            publishedAt,
            url
        };
    }

    public render() {
        return (
            <div className="article">
                <a href={ this.state.url }>
                  <h3>{ this.state.headline }</h3>
                </a>
                <p>{ this.state.publishedAt }</p>
                <p>{ this.state.articleBody }</p>
            </div>
        );
    }
}

export default Article;
