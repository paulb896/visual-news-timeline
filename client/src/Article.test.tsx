import * as React from 'react';
import * as Renderer from 'react-test-renderer';
import Article from './Article';

test('Article renders correctly', () => {
  const tree = Renderer
    .create(<Article 
        key={ "unique-article-id" }
        id={ "unique-article-id" }
        articleBody={ "At five seconds past midnight, Ian Power became the owner of the first gram of recreational cannabis to be sold in Canada. After waiting for hours in the cold, Power strode excitedly into the only cannabis store open in St Johnâ€™s, Newfoundland, and made his purchase." }
        publishedAt={ "2018-10-17T18:41:34Z" }
        url={ "https://www.theguardian.com/world/2018/oct/17/on-a-high-canada-celebrates-cannabis-being-legalised" }
        headline={ "On a high: Canada celebrates cannabis being legalised" } />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});