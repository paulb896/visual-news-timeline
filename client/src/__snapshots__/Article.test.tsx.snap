// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div
  className="article"
>
  <a
    href="https://www.theguardian.com/world/2018/oct/17/on-a-high-canada-celebrates-cannabis-being-legalised"
  >
    <h3>
      On a high: Canada celebrates cannabis being legalised
    </h3>
  </a>
  <p>
    2018-10-17
     - 
    18:41:34
  </p>
  <p>
    At five seconds past midnight, Ian Power became the owner of the first gram of recreational cannabis to be sold in Canada. After waiting for hours in the cold, Power strode excitedly into the only cannabis store open in St Johnâ€™s, Newfoundland, and made his purchase.
  </p>
</div>
`;
